clippy_lints/src/disallowed_macros.rs

```diff
@@ -125,15 +120,83 @@ impl DisallowedMacros {
 }
 
 impl_lint_pass!(DisallowedMacros => [DISALLOWED_MACROS]);
-
 impl LateLintPass<'_> for DisallowedMacros {
-    fn check_crate(&mut self, cx: &LateContext<'_>) {
-        // once we check a crate in the late pass we can emit the early pass lints
-        if let Some(attr_spans) = self.early_macro_cache.clone().0.get() {
-            for span in attr_spans {
-                self.check(cx, *span, None);
-            }
-        }
+    fn check_attribute(&mut self, cx: &LateContext<'_>, attr: &Attribute) {
+        let span = match attr {
+            Attribute::Unparsed(attr_item) => attr_item.span,
+            Attribute::Parsed(kind) => match kind {
```

Comment:
I am not particularly happy with this giant match block.
There is [a method](https://github.com/rust-lang/rust/blob/7f7b8ef27d86c865a7ab20c7c42f50811c6a914d/compiler/rustc_hir/src/hir.rs#L1302) that is perhaps meant to return the `Attribute`'s `Span`, but it is notcomplete, and panics for the attributes that do not have one.  It would make sense for this match to move next to that `span()` method if this is the right approach at all.